src/replication.c

```diff
@@ -3215,10 +3228,14 @@ void replicationSendAck(void) {
 
     if (c != NULL) {
         c->flags |= CLIENT_MASTER_FORCE_REPLY;
-        addReplyArrayLen(c,3);
+        addReplyArrayLen(c,server.fsynced_reploff != -1 ? 5 : 3);
```

Comment:
Maybe `server.fsynced_reploff >= 0` would be safer/clearer?

Reply:
i think it's more explicit this way, showing that -1 has a special meaning.